src/workflows/transport/common_transport.py

```diff
@@ -2,13 +2,18 @@
 
 import decimal
 import logging
-from typing import Any, Callable, Dict, Mapping, Optional, Set
+from typing import Any, Callable, Dict, Mapping, NamedTuple, Optional, Set
 
 import workflows
 
 MessageCallback = Callable[[Mapping[str, Any], Any], None]
 
 
+class TemporarySubscription(NamedTuple):
```

Comment:
Yes, typing.NamedTuple is a typed named tuple. One benefit over dataclasses here is that it allows unpacking of the queue name, but otherwise the reason I picked namedtuples is because that's what I picked.

Reply:
I guess allowing tuple-unpacking of the queue name could also be a negative, depending on your point of view :)